packages/webapp/src/containers/InviteSignUp/index.jsx

```diff
@@ -68,37 +73,76 @@ function InviteSignUp({ history }) {
+    onSuccess: async (tokenResponse) => {
+      try {
+        // I have not previously used redux or redux-saga, but I imagine a query to an API like this should be in a saga.js file?
+        const userInfo = await axios.get('https://www.googleapis.com/oauth2/v3/userinfo', {
+          headers: { Authorization: `Bearer ${tokenResponse.access_token}` },
+        });
+        onSuccessGoogle(userInfo.data, tokenResponse);
+      } catch (error) {
+        console.log(error);
+      }
+    },
+    onError: (errorResponse) => console.log(errorResponse),
+  });
+
+  /* Shape of response object from hook:
+
+      { 
+        access_token: ...,
+        authuser: "2",
+        expires_in: 3599
+        prompt: "none",
+        scope: "email profile ...",
+        token_type: "Bearer"
+      }
+
+
+    Shape of response from https://www.googleapis.com/oauth2/v3/userinfo using the access_token above:
+
+      { 
+        email: ...
+        email_verified: true,
+        family_name: ...
+        given_name: ...
+        locale: ...,
+        name: ...
+        picture: <link>
+        sub: ...
+      }
+
+  */
+
+  /* ProceedButton is a wrapper for the renderProps that handles the onClick similar to how the previous package's GoogleLogin component did: https://github.com/anthonyjgrove/react-google-login/blob/master/src/google-login.js I wrote it like this in order to keep the surrounding code unchanged. I don't really use/understand renderProps myself so I will defer to someone else if maybe they can be refactored out, now that our new package doesn't require them. */
+  const ProceedButton = ({ render }) => {
+    return render({ onClick });
+  };
 
   return (
-    <>
-      <PureInviteSignup
-        googleButton={
-          <GoogleLogin
-            render={(renderProps) => (
-              <Button
-                data-cy="invitedUser-proceed"
-                onClick={onClickGoogle(renderProps)}
-                disabled={(selectedKey === GOOGLE && renderProps.disabled) || !selectedKey}
-                fullLength
-              >
-                {t('common:PROCEED')}
-              </Button>
-            )}
-            onSuccess={onSuccessGoogle}
-            onFailure={onFailureGoogle}
-            clientId={clientId}
-          >
-            {t('SIGNUP.GOOGLE_BUTTON')}
-          </GoogleLogin>
-        }
-        showError={showError}
-        selectedKey={selectedKey}
-        email={email}
-        onClick={onClick}
-        isChrome={isChrome()}
-      />
-    </>
+    <PureInviteSignup
+      googleButton={
+        <ProceedButton
+          render={(renderProps) => (
+            <Button
+              data-cy="invitedUser-proceed"
+              onClick={onClickProceed(renderProps)}
+              disabled={(selectedKey === GOOGLE && renderProps.disabled) || !selectedKey}
+              fullLength
+            >
+              {t('common:PROCEED')}
+            </Button>
+          )}
+        />
+      }
```

Comment:
If we don't need to wait for someone to refactor this...

I believe this `renderProps` was previously passed from the `GoogleLogin` component and is no longer necessary. I also believe that `renderProps`(line 128) is the `onClick` that you are passing in line 121, so the `googleButton` can be:
```
  googleButton={
    <Button
      data-cy="invitedUser-proceed"
      onClick={onClickProceed}
      disabled={!selectedKey}
      fullLength
    >
      {t('common:PROCEED')}
    </Button>
  }
/>
```

`onClickProceed`(line 61) does not need to return a function anymore, and `renderProps.onClick();` can be removed because what actually is happening is `setSelectedKey();`

Reply:
THANK YOU!!! My head was spinning with all the functions being passed around and this is the simplification that I hoped existed 😄 So much better!!!